src/audio/host-zephyr.c

```diff
@@ -1027,8 +1027,7 @@ static int host_position(struct comp_dev *dev,
 void host_zephyr_reset(struct host_data *hd, uint16_t state)
 {
 	if (hd->chan) {
-		if (state == COMP_STATE_ACTIVE)
-			dma_stop(hd->chan->dma->z_dev, hd->chan->index);
+		dma_stop(hd->chan->dma->z_dev, hd->chan->index);
```

Comment:
Does this work in all states, including if reset is called twice in a row with no intervening calls? (just a question since I'm not fully familiar with this)

Reply:
@paulstelian97 I think this is ok, Zephyr dma.h has for stop that "stop is allowed on channels that have already been stopped and must report success. " ... the documentation was recently clarified on Zephyr side as SOF had multiple places where we could potentially have redundant calls. But we save a few bytes and cycles if we don't need to track state in both SOF and Zephyr...

Reply:
The question here is that does stop work even after the prior dma_release_channel?

But I think I can answer that myself (yes, because hd->chan will be null and this code path won't run anyway)